lib/instance.js

```diff
@@ -236,15 +236,14 @@ module.exports = (function() {
 
         if (options.attributes) {
           keys = options.attributes;
-          if (this.Model._hasVirtualAttributes) {
-            keys = keys.concat(this.Model._virtualAttributes);
-          }
 
           if (this.options.includeNames) {
             keys = keys.concat(this.options.includeNames);
+          } else if (this.Model._hasVirtualAttributes) {
+            keys = keys.concat(this.Model._virtualAttributes);
           }
 
-          for (i = 0, length = keys.length; i < length; i++) {
+          for (i = 0; i < keys.length; i++) {
```

Comment:
http://jsperf.com/forloops3/41
This is even the case for _really_ old V8 versions that would ship with node < 0.8